src/main/kotlin/operators/TreepatFunction.kt

```diff
@@ -4,8 +4,19 @@ fun treepatFunction(expression: VisitorFunction): VisitorFunction = { targetTree
     var currentNode = targetTreeNode
     val answers = mutableListOf<VisitorFunctionResponse>()
     while (currentNode != null) {
-        answers.add(expression.invoke(currentNode))
+        val response = expression.invoke(currentNode)
+        answers.add(
+            VisitorFunctionResponse(
+                response.responses.filter { it.matches.isNotEmpty() },
+                response.hasMatch
+            )
+        )
         currentNode = currentNode.nextLeftmostPreorderNode()
     }
-    VisitorFunctionResponseFactory.createResponse(answers, targetTreeNode)
+    val allHasMatches = answers.filter { it.hasMatch }.filter { it.responses.isNotEmpty() }.map { it.responses.last() }
```

Comment:
Get last valid matches, is the bigger match in breadth closure. Check test case 8 in the breadth close folder.